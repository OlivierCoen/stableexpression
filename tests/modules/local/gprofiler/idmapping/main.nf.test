nextflow_process {

    name "Test Process GPROFILER_IDMAPPING"
    script "modules/local/gprofiler/idmapping/main.nf"
    process "GPROFILER_IDMAPPING"
    tag "idmapping"
    tag "module"

    test("Map Ensembl IDs to themselves") {

        when {
            process {
                """
                meta = []
                count_file = file("$projectDir/tests/data/idmapping/base/counts.ensembl_ids.csv")
                input[0] = [meta, count_file, "Solanum tuberosum"]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Map NCBI IDs") {

        when {
            process {
                """
                meta = []
                count_file = file("$projectDir/tests/data/idmapping/base/counts.ncbi_ids.csv")
                input[0] = [meta, count_file, "Arabidopsis thaliana"]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }


    test("Map Uniprot IDs") {

        when {
            process {
                """
                meta = []
                count_file = file("$projectDir/tests/data/idmapping/base/counts.uniprot_ids.csv")
                input[0] = [meta, count_file, "Arabidopsis thaliana"]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Empty count file - ignore error") {

        tag "idmapping_empty"

        when {
            process {
                """
                meta = []
                count_file = file("$projectDir/tests/data/idmapping/empty/counts.csv")
                input[0] = [meta, count_file, "Arabidopsis thaliana"]
                """
            }
        }

        then {
            assert process.success
            assert process.trace.succeeded().size() == 0
            assert process.trace.failed().size() == 1
            // check for the absence of expected output (the error is ignored but no output is produced)
            assert process.out.renamed.size() == 0
            assert process.out.metadata.size() == 0
            assert process.out.mapping.size() == 0
        }

    }

    test("Mapping not found - ignore error") {

        tag "idmapping_not_found"

        when {
            process {
                """
                meta = []
                count_file = file("$projectDir/tests/data/idmapping/not_found/counts.csv")
                input[0] = [meta, count_file, "Homo sapiens"]
                """
            }
        }

        then {
            assert process.success
            assert process.trace.succeeded().size() == 0
            assert process.trace.failed().size() == 1
            // check for the absence of expected output (the error is ignored but no output is produced)
            assert process.out.renamed.size() == 0
            assert process.out.metadata.size() == 0
            assert process.out.mapping.size() == 0
        }

    }

}
