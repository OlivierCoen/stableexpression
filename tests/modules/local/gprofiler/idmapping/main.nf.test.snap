{
    "Map Ensembl IDs to themselves": {
        "content": [
            {
                "0": [
                    "counts.ensembl_ids_renamed.csv:md5,ef96059e3283d4305b2c004d649ae648"
                ],
                "1": [
                    "counts.ensembl_ids_metadata.csv:md5,608af2e5f8df9d8d2bb5829c04d6f9eb"
                ],
                "2": [
                    "counts.ensembl_ids_mapping.csv:md5,a2d01dbbe84a30640825befdda3e1963"
                ],
                "3": [
                    [
                        "GPROFILER_IDMAPPING",
                        "python",
                        "3.13.0"
                    ]
                ],
                "4": [
                    [
                        "GPROFILER_IDMAPPING",
                        "pandas",
                        "2.2.3"
                    ]
                ],
                "5": [
                    [
                        "GPROFILER_IDMAPPING",
                        "requests",
                        "2.32.3"
                    ]
                ],
                "metadata": [
                    "counts.ensembl_ids_metadata.csv:md5,608af2e5f8df9d8d2bb5829c04d6f9eb"
                ],
                "renamed": [
                    "counts.ensembl_ids_renamed.csv:md5,ef96059e3283d4305b2c004d649ae648"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-03T15:43:54.267190946"
    },
    "Map Uniprot IDs": {
        "content": [
            {
                "0": [
                    "counts.uniprot_ids_renamed.csv:md5,a93d7145f8b35f6074cdf21c7c97de7c"
                ],
                "1": [
                    "counts.uniprot_ids_metadata.csv:md5,b87d6533848a3ae07b289ec6b0c4a1ff"
                ],
                "2": [
                    "counts.uniprot_ids_mapping.csv:md5,fe88c79c45d45825d28f325f7a2f383e"
                ],
                "3": [
                    [
                        "GPROFILER_IDMAPPING",
                        "python",
                        "3.13.0"
                    ]
                ],
                "4": [
                    [
                        "GPROFILER_IDMAPPING",
                        "pandas",
                        "2.2.3"
                    ]
                ],
                "5": [
                    [
                        "GPROFILER_IDMAPPING",
                        "requests",
                        "2.32.3"
                    ]
                ],
                "metadata": [
                    "counts.uniprot_ids_metadata.csv:md5,b87d6533848a3ae07b289ec6b0c4a1ff"
                ],
                "renamed": [
                    "counts.uniprot_ids_renamed.csv:md5,a93d7145f8b35f6074cdf21c7c97de7c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-03T15:44:24.705004112"
    },
    "Map NCBI IDs": {
        "content": [
            {
                "0": [
                    "counts.ncbi_ids_renamed.csv:md5,a93d7145f8b35f6074cdf21c7c97de7c"
                ],
                "1": [
                    "counts.ncbi_ids_metadata.csv:md5,b87d6533848a3ae07b289ec6b0c4a1ff"
                ],
                "2": [
                    "counts.ncbi_ids_mapping.csv:md5,fe4fd9005dce99b7722b84134f51badd"
                ],
                "3": [
                    [
                        "GPROFILER_IDMAPPING",
                        "python",
                        "3.13.0"
                    ]
                ],
                "4": [
                    [
                        "GPROFILER_IDMAPPING",
                        "pandas",
                        "2.2.3"
                    ]
                ],
                "5": [
                    [
                        "GPROFILER_IDMAPPING",
                        "requests",
                        "2.32.3"
                    ]
                ],
                "metadata": [
                    "counts.ncbi_ids_metadata.csv:md5,b87d6533848a3ae07b289ec6b0c4a1ff"
                ],
                "renamed": [
                    "counts.ncbi_ids_renamed.csv:md5,a93d7145f8b35f6074cdf21c7c97de7c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-03T15:44:09.676405952"
    }
}